pygccxml/parser/scanner.py

```diff
@@ -154,6 +154,11 @@ def __init__(self, xml_file, decl_factory, config, *args):
         self.__mangled_suffix = ' *INTERNAL* '
         self.__mangled_suffix_len = len(self.__mangled_suffix)
 
+        self.__name_attrs_to_skip = []
+        self.__read_location = \
+            lambda decl, attrs, to_skip: self.__read_location_bootstrap(
```

Comment:
It's not that bad. Of course it's not ideal, and does make the code more readable; but if it helps for the performance, it needs to be done. I'll merge it as it.